src/models/quote.js

```diff
@@ -1,38 +1,29 @@
 'use strict'
 
-const request = require('co-request')
 const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 const UnacceptableAmountError = require('../errors/unacceptable-amount-error')
 const AssetsNotTradedError = require('../errors/assets-not-traded-error')
-const ExternalError = require('../errors/external-error')
 const DEFAULT_DESTINATION_EXPIRY = 5 // seconds
 
 function * makeQuoteQuery (params) {
-  const sourceLedger = params.source_ledger ||
-    (yield getAccountLedger(params.source_account))
-  const destinationLedger = params.destination_ledger ||
-    (yield getAccountLedger(params.destination_account))
+  const sourceLedger = addressToLedger(params.source_address)
+  const destinationLedger = addressToLedger(params.destination_address)
   const explain = params.explain === 'true'
   return {
     sourceLedger,
     destinationLedger,
     sourceAmount: params.source_amount,
     destinationAmount: params.destination_amount,
+    destinationPrecisionAndScale: params.destination_precision && {
+      precision: params.destination_precision,
+      scale: params.destination_scale
+    },
     explain
   }
 }
 
-function * getAccountLedger (account) {
-  const res = yield request({
-    method: 'get',
-    uri: account,
-    json: true
-  })
-  const ledger = res.body && res.body.ledger
-  if (res.statusCode !== 200 || !ledger) {
-    throw new ExternalError('Unable to identify ledger from account: ' + account)
-  }
-  return ledger
+function addressToLedger (address) {
+  return address.split('.').slice(0, -1).join('.')
```

Comment:
> So where does the low-level address come from originally? If at the application layer the user enters an SPSP address like alice@red.ilpdemo.org, how does the application generate us.fed.wf.alice?

SPSP is described in [IL-RFC-0009](https://github.com/interledger/rfcs/blob/master/0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md).

You first query webfinger to resolve the email-like identifier:

https://red.ilpdemo.org/.well-known/webfinger?resource=acct%3Abob%40red.ilpdemo.org

That tells us the SPSP endpoint URI `https://red.ilpdemo.org/api/receivers/bob`.

https://red.ilpdemo.org/api/receivers/bob

Querying that tells us the ILP address. Right now that's a `ledger, account` tuple, e.g. `ledger: 'https://red.ilpdemo.org/ledger', account: 'https://red.ilpdemo.org/ledger/accounts/alice'`. If this pull request lands, that would change to just be an ILP address, e.g. `ilpdemo.red.alice`.

If you're asking how the SPSP server (=receiver) knows its own address - right now that's just part of the SPSP server configuration. In the future we will likely want to come up with some auto-configuration scheme but that's out of scope for this PR.